tests/providers/amazon/aws/hooks/test_emr.py

```diff
@@ -191,6 +197,9 @@ def test_get_cluster_id_by_name(self):
 
         assert no_match is None
 
+        with pytest.raises(AirflowException):
+            hook.get_cluster_id_by_name("test_cluster0", ["RUNNING", "WAITING", "BOOTSTRAPPING"])
+
```

Comment:
Should you not add a test case which covers the issue this PR is solving? I.e. A success case for getting a cluster id for a cluster that would be on the second page.

Reply:
I'm not sure if list-clusters API on EMR is ordered at all, so I don't know if there's a way to guarantee that a given test-case cluster shows up on the second page only. If there's a way to guarantee this, I can add that as a test.

Reply:
Their API [docs](https://docs.aws.amazon.com/emr/latest/APIReference/API_ListClusters.html) says unordered list. I could change the internal logic of the function to order the cluster, but that feels silly to do just to make a test.

Reply:
I dont think that's what @o-nikolas is saying. I think he wants to have a test to cover the use case **in case** the cluster you are looking for is in the second page. The implementation should not change, just adding a test that covers this use case would increase the testing coverage. And I agree with that

Reply:
I totally understand, I just am not sure how to force a given clusterName to be on the second page. Can I assume that even if the listClusters API provides an unordered response, it will provide that same order in subsequent queries? Then I can do a regular listClusters and fetch a cluster on the second page. But if the ordering is not consistent, I don't really have a surefire way to guarantee that the clusterName is on the second page. I could increase the number of clusters (<1000) to make the probability very high.

I'm happy to add tests, even for testing duplicate clusterNames which wasn't part of this PR. I'm just not too familiar with the EMR API, so I'm looking for some guidance.

Reply:
I just realized with the way the marker works, that listClusters will be consistent. I'll edit the test accordingly.

Reply:
Addressed